lib/alerts/medicine_article_monitor.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+# This class to create alerts when a WikiProject Medicine article
+# is assigned for a course that doesn't have the medical content
+# training module.
+# wiki_education mode only
+# Goal: to catch cases where a class that wasn't expected 
+# to work on medical topics tries to do so.
+# Issue #4693
+class MedicineArticleMonitor
+  WIKI_PROJECT_MEDICINE_CAT = 'Category:All WikiProject Medicine articles'
+  NUMBER_OF_ARTICLES_TO_CHECK = 5
+
+  def self.create_alerts_for_no_med_training_for_course
+    new.create_alerts_for_no_med_training_for_course
+  end
+
+  def initialize
+    @wiki = Wiki.find_by(language: 'en', project: 'wikipedia')
+  end
+
+  def create_alerts_for_no_med_training_for_course
+    last_assigned_med_articles.each do |assignment|
+      unless med_training_for_course?(assignment.course_id)
+        create_alert_for_no_med_training_for_course(assignment)
+      end
+    end
+  end
+
+  def create_alert_for_no_med_training_for_course(assignment)
+    return if alert_already_exists?(assignment)
+
+    alert = Alert.create!(type: 'NoMedTrainingForCourseAlert',
+                  article_id: assignment.article_id,
+                  course_id: assignment.course_id)
+    alert.email_content_expert
+  end
+
+  def alert_already_exists?(assignment)
+    Alert.exists?(type: 'NoMedTrainingForCourseAlert',
+                  article_id: assignment.article_id,
+                  course_id: assignment.course_id)
+  end
+
+  def last_assigned_med_articles
+    last_assigned_articles
+      .select { |assignment| med_article?(assignment.article_title) }
+  end
+
+  def last_assigned_articles
+    Assignment.last NUMBER_OF_ARTICLES_TO_CHECK
+  end
+
+  def med_training_for_course?(course_id)
```

Comment:
This can be done a little bit more efficiently and concisely. The medical topics training actually has two variants for different kinds of courses, but these modules have hard-coded IDs (11, 48).

You can make a constant for the array of medical training IDs, and check whether either of them are found among the `training_module_ids` for a course (ie, `(Course.find(course_id).training_module_ids & MEDICAL_TRAINING_IDS).any?` or similar... that can probably be refactored a bit for clarity).